core/math/camera_matrix.cpp

```diff
@@ -84,6 +84,9 @@ void CameraMatrix::adjust_perspective_znear(real_t p_new_znear) {
 }
 
 void CameraMatrix::set_perspective(real_t p_fovy_degrees, real_t p_aspect, real_t p_z_near, real_t p_z_far, bool p_flip_fov) {
+	// Zero or negative Z near values are only valid in orthogonal projection.
+	p_z_near = MAX(0.01, p_z_near);
```

Comment:
This is a fairly far away limit for the near plane. This means nothing will be visible any closer than 10mm to the camera. We should have this hard limit be much smaller, perhaps 0.0001 (0.1mm) or at least 0.001 (1mm). Even though most users will want a smaller value, we shouldn't disallow it with a high hard-coded limit.

Reply:
I agree, this shouldn't be a hard-coded limit, even if the chances of someone needing that small of a near are very tiny.

Reply:
I changed the lower limit to 0.001.